File: 2236-maximum-twin-sum-of-a-linked-list/maximum-twin-sum-of-a-linked-list.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    
    public int pairSum(ListNode head) {
        int[] nums = new int[100000];

        ListNode node = head;

        int index = 0;
        while (node != null) {
            nums[index] = node.val;
            node = node.next;
            index++;
        }

        int result = Integer.MIN_VALUE;

        for (int i = 0; i < index / 2; i++)
            result = Math.max(result, nums[i] + nums[index - i - 1]);

        return result;
    }

}

File: 2236-maximum-twin-sum-of-a-linked-list/maximum-twin-sum-of-a-linked-list.c
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     struct ListNode *next;
 * };
 */
int pairSum(struct ListNode* head) {
    int values[100000];
    int valuesSize = 0;
    struct ListNode* node = head;
    while (node) {
        values[valuesSize++] = node->val;
        node = node->next;
    }
    int valuesSize2 = valuesSize / 2;
    int valuesSize1 = valuesSize - 1;
    int result = 0;
    for (int i = 0; i < valuesSize2; i++) {
        int sum = values[i] + values[valuesSize1 - i];
        if (result < sum) result = sum;
    }
    return result;
}